code/game/objects/structures/holosign.dm

```diff
@@ -104,6 +104,72 @@
 	. = ..()
 	air_update_turf(TRUE)
 
+///this is a machinery to make it possible to check on the APC power to shut it down in case of an outage
+/obj/machinery/holosign/barrier/power_shield
+	name = "powered shield"
+	desc = "A shield to prevent changes of atmospheric and heat transfer"
+	icon = 'icons/effects/effects.dmi'
+	density = FALSE
+	anchored = TRUE
+	CanAtmosPass = ATMOS_PASS_NO
+	resistance_flags = FIRE_PROOF
+	///store the conductivity value of the turf is applyed so that it can be restored on removal
+	var/stored_conductivity = 0
+	///power drain from the apc, in W (so 5000 is 5 kW), per each holosign placed
+	var/power_consumption = 5000
+	var/obj/item/holosign_creator/shield_projector
+
+/obj/machinery/holosign/barrier/power_shield/Initialize(loc, source_projector)
+	. = ..()
+	var/area/a = get_area(src)
+	if(a.power_equip == FALSE)
+		stack_trace("Power shield created without power avaiable")
+		qdel(src)
+		return
+	if(source_projector)
+		shield_projector = source_projector
+		shield_projector.signs += src
+	air_update_turf(TRUE)
+	a.addStaticPower(power_consumption, STATIC_EQUIP)
+	shield_turf()
+
+/obj/machinery/holosign/barrier/power_shield/Destroy()
+	var/turf/T = loc
+	T.thermal_conductivity = stored_conductivity
+	var/area/a = get_area(src)
+	a.addStaticPower(-power_consumption, STATIC_EQUIP)
+	if(shield_projector)
+		shield_projector.signs -= src
+		shield_projector = null
```

Comment:
Can the `shield_projector` survive the destruction of the `power_shield`?

Reply:
Yes, why?